src/containers/ActionMenuContainer/ActionMenuContainer.tsx

```diff
@@ -124,7 +124,11 @@ export const ActionMenuContainer: React.FC = () => {
       onMint={(amount, decimal) => () => {
         dispatch(actions.mint({ amount: amount.mul(new BN(10 ** 6)).div(new BN(10 ** decimal)) }))
       }}
+      onWarning={() => () => {
```

Comment:
I think that you can move modal there, show it on click in repay tab and use current onBurn as confirmation callback in modal instead of adding new props

Reply:
done